### ISSUE TITLE
options parameter statement_timeout doesn't work with pgbouncer 

### ISSUE THREAD

**Issue by oceancode11** [pgbouncer/pgbouncer#525]
I am unable to get statement_timeout working with pgbouncer when using `PGOPTIONS` /`options`. I added `options` to `ignore_startup_parameters` but its clear if pgbouncer is passing these or completely ignoring. 

This is the config
```
...
[pgbouncer]
pool_mode = transaction
ignore_startup_parameters = options
...
```

Using psql
```
env PGOPTIONS="-c statement_timeout=1000" psql -h <host> -d <dbname> -U <user> -W 
```

The result I get. 
```
=> show statement_timeout;
 statement_timeout 
-------------------
 0
(1 row)
```

I expect it to be 1000ms. Thank you!

**Comment by oceancode11** [comment]
This what I see in DEBUG logs. So i guess its not possible then?

```
ignoring startup parameter: options=-c statement_timeout=1000
```

**Comment by oceancode11** [comment]
This is what i found from https://github.com/Athou/commafeed/issues/559#issuecomment-382106124

> http://lists.pgfoundry.org/pipermail/pgbouncer-general/2008-November/000307.html

> There are few parameters that pgbouncer can keep track of.
Like client_encoding. Because postgres reports back chenges
of that parameter. The 'extra_float_digits' is not one of them.
Thus, in transaction pooling mode, there is no way to give
consistent picture of that parameter to clients.

> Giving hard-error instead of ignoring by default is delierate,
so admin can decide if the parameter is actually important
for application logic. If it is then you should make
if default from postgres config.

So pgbouncer cant' keep track of `options` or similar params and hence doesn't fail hard but ignores. Is it possible to consider this as a feature request?

**Comment by ilannn** [comment]
I also came across this issue and was really frustrated. I understand the reason why transaction pooling may be hard to implement, but since the session carried this information maybe we could use it per transaction?

**Comment by pablogamboa** [comment]
@JelteF could this be issue considered as fixed since 1.20.0 came out?

**Comment by JelteF** [comment]
In a sense yes, you could add it `statement_timeout` to `track_extra_parameters` but you'll still have to install a postgres extension that sets the GUC_REPORT for the `statement_timeout` to actually track it correctly (and currently there's no such postgres extension).

**Comment by pablogamboa** [comment]
Thanks for clarifying @JelteF. I don't think this issue can be considered as fixed yet. Looking forward to PG implementing it! Most of us are running managed databases with limited options to load extensions.

I guess that the best way to have custom `statement_timeout`'s is to set them at the role level if you want a pgbouncer-friendly way of customizing it. Can you confirm @JelteF for me and future readers? 😄

**Comment by JelteF** [comment]
Yes, setting statement_timeout at the role level would work. You could also have pgbouncer set `statement_timeout` for you like using `connect_query`:
```
[databases]
mydb = host=somehost connect_query='set statement_timeout=5s;'

**Comment by monishs1201** [comment]
Hi, I'm trying to use PgBouncer with Citus read scaling via:

PGOPTIONS="-c citus.use_secondary_nodes=always"

Direct PostgreSQL connections work correctly:

psql -h <postgres> ...
SHOW citus.use_secondary_nodes;
-- returns 'always'

But through PgBouncer:

psql -h <pgbouncer> ...

I receive either:
FATAL: unsupported startup parameter in options: citus.use_secondary_nodes

or, when using ignore_startup_parameters = options

the parameter is silently ignored and

 SHOW citus.use_secondary_nodes;
returns:never

Important distinction from statement_timeout

I understand previous discussions around:

statement_timeout
track_extra_parameters
connect_query
ALTER ROLE SET

However, citus.use_secondary_nodes appears fundamentally different because it is a startup/backend-level GUC (PGC_BACKEND) and cannot be changed after backend initialization.

For example:

SET citus.use_secondary_nodes='always';

fails with:

ERROR: parameter "citus.use_secondary_nodes" cannot be set after connection start

Likewise:

ALTER ROLE myuser SET citus.use_secondary_nodes='always';

also fails for the same reason.

So unlike statement_timeout, neither:

role-level settings
connect_query
session SET
transaction SET

can solve this.

My question

Is there currently ANY supported mechanism in PgBouncer to:

accept arbitrary startup options=-c ...
preserve them during backend creation
forward them to PostgreSQL during the initial connection handshake
especially for startup-only (PGC_BACKEND) GUCs

when using pool_mode=session?

Or is this fundamentally unsupported by PgBouncer's architecture today? @JelteF @pablogamboa
